code/_onclick/click.dm

```diff
@@ -145,6 +145,11 @@
 		return
 
 	//Standard reach turf to turf or reaching inside storage
+
+	//This is sent regardless of range and is primarily used for 3x3 runes so that you can use them while not standing one them.
+	SEND_SIGNAL(A, COMSIG_ATOM_ATTACKED_RANGE_IGNORE, src)
```

Comment:
Signals are supposed to be generic event triggers. This one is a pretty snowflake one to fit in a particular situation.
If you want the rune to be triggered regardless or reach you can register to multiple signals, such as in UnarmedAttack, RangedAttack and the melee_attack_chain.
That is better than adding a signal for every particular situation, that causes unnecessary bloat.
Unless if you can think of this as a generic event that's being reported, but it doesn't look so to me.

Reply:
i dont think youve looked closely enoigh at the code. all the signals youve listed are sent to thr mob not the atom. afaik this is the only signal that is sent to the atom upon being clicked

Reply:
Then add a signal in the same way as `afterattack` does it, reporting the generic event instead of the specific conditions that meet your current feature's circumstance.

Reply:
I mean this is a generic event of an atom being clicked, the name is misleading i will rename it

Reply:
If you want to report that generic event, then put it next to `COMSIG_MOB_CLICKON`, right afterwards.